templates/ruby/lib/container/payload.rb.twig

```diff
@@ -0,0 +1,77 @@
+module Appwrite
+  class Payload
+    attr_reader :data
+    attr_reader :filename
+
+    def initialize(data: nil, path: nil, filename: nil)
+      @path = path
+      @data = data
+      @filename = filename
+
+      @size = if @data then
+        @data.bytesize
+      else
+        File.size(@path)
+      end
+    end
+
+    def read(offset: 0, length: nil)
+      if @data then
+        @data.byteslice(offset, length || @data.bytesize)
+      else
+        IO.read(@path, length || File.size(@path), offset)
+      end
+    end
+
+    # @param [String] path
+    # @param [String, nil] filename
+    # @return [Payload]
+    def self.from_file(path, filename: nil)
+        filename = if filename.nil? then
+          File.basename(path)
+        else
+          filename
+        end
+      new(nil, path, filename)
```

Comment:
Hmm, data nil and we pass file?
Can we instead simply get data here, and pass it? That way constructor would probably not need path anymore.

Reply:
In the case of large files, you load everything into memory this way.

We can do it, but it's not how it was before